indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,265 @@
+            # Class already exists
+            if update:
+                updated_class = self.create_class_template(issuer_id, class_suffix)
+                response = self.http_client.put(url=f'{self.class_url}/{issuer_id}.{class_suffix}', json=updated_class)
+                if response.status_code != 200:
+                    raise BadRequest(response.text)  # Something else went wrong...
+            return response.json()
+        elif response.status_code != 404:
+            Logger.get('events').warning('Cannot create Google Wallet class: %s', response.text)
+            raise BadRequest(response.text)  # Something else went wrong...
+
+        # See link below for more information on required properties
+        # https://developers.google.com/wallet/tickets/events/rest/v1/eventticketclass
+        new_class = self.create_class_template(issuer_id, class_suffix)
+        response = self.http_client.post(url=self.class_url, json=new_class)
+
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise BadRequest(response.text)
+
+    def patch_class(self, issuer_id: str, class_suffix: str, patch_body: dict) -> Optional[str]:
+        """Patch a class.
+
+        The PATCH method supports patch semantics.
+        """
+        # Check if the class exists
+        response = self.http_client.get(url=f'{self.class_url}/{issuer_id}.{class_suffix}')
+
+        if response.status_code == 200:
+            response = self.http_client.patch(url=f'{self.class_url}/{issuer_id}.{class_suffix}', json=patch_body)
+            return response.json()
+        elif response.status_code == 404:
+            return None  # Class does not exist.
+        else:
+            raise BadRequest(response.text)  # Something else went wrong...
+
+    def create_object_template(self, issuer_id: str, class_suffix: str, object_suffix: str) -> dict:
+        """This method will return a dict format ticket object template."""
+        from indico.modules.events.registration.util import get_persons, get_qr_data
+        qr_data = get_qr_data(get_persons([self.registration])[0])
+        dict_template = {
+            'id': f'{issuer_id}.{object_suffix}',
+            'classId': f'{issuer_id}.{class_suffix}',
+            'state': 'ACTIVE',
+            'barcode': {
+                'type': 'QR_CODE',
+                'value': str(qr_data)
+            },
+            'hexBackgroundColor': '#007cac',
+            'textModulesData': [
+                {
+                    'header': 'Name',
+                    'body': self.registration.full_name,
+                    'id': 'namefield'
+                },
+                {
+                    'header': 'Email',
+                    'body': self.registration.email,
+                    'id': 'emailfield'
+                }
+            ],
+            'ticketHolderName': self.registration.full_name,
+            'ticketNumber': self.registration.id
```

Comment:
Maybe this should be the `friendly_id`? Or `f'#{self.registration.friendly_id}'` since we show it with the `#` in other places as well?